pippy/PipelineSchedule.py

```diff
@@ -464,27 +529,32 @@ def step_microbatches(
                     if ops:
                         dist.batch_isend_irecv(ops)
 
-    def step(self, *args, **kwargs):
-        # TODO
-        pass
 
-
-class ScheduleInterleaved1F1B(PipelineSchedule):
-    def __init__(self, stages: List[PipelineStageBase]):
-        if len(stages) <= 1:
+class ScheduleInterleaved1F1B(PipelineScheduleMulti):
+    def __init__(
+        self,
+        stages: List[PipelineStageBase],
+        n_microbatches: int,
+        loss_fn: Optional[Callable] = None,
+        output_merge_spec: Optional[Union[Dict[str, Any], Tuple[Any]]] = None,
+    ):
+        self.pp_group_size = stages[0].group_size
+        # TODO: is this limitation a must?
+        if n_microbatches % self.pp_group_size != 0:
             raise ValueError(
-                "Looped DFS schedule requires at least two stages to be used."
+                f"Interleaved 1F1B schedule requires the number of microbatches ({self._n_microbatches}) \
+                to be a multiple of the number of pipeline ranks ({self.pp_group_size})."
             )
```

Comment:
@H-Huang wondering if this is a real limitation. If it is, can you put some words here to describe the reason?

Reply:
i dont think this is a hard requirement, but i haven't seen the literature describe otherwise. I have to do some tests and think through to see if different sized microbatches also works